### PR TITLE
[10.0] porting l10n_it_fatturapa*

### PR DESCRIPTION
**Pull request by eLBati**
Depends on https://github.com/OCA/l10n-italy/pull/288 (merged)

### CODE REVIEW

**Comment by Giuliano69** [comment]
Non riesco a capire cosa faccia fallire il test su runbot ... non mi sembra che il dettaglio dia informazioni utili.
Come si fa a capire cosa fallisce in questi casi ?

**Comment by eLBati** [comment]
@Giuliano69 nella pagina https://runbot.odoo-community.org/runbot/build/3277459 devi cliccare sulla rotella e poi ```full all logs```
In questo caso: ```module l10n_it_fatturapa_out: Unmet dependencies: l10n_it_split_payment```

**Comment by tafaRU** [comment]
@eLBati, please rebase on top to include https://github.com/OCA/l10n-italy/pull/288

**Comment by eLBati** [comment]
About pylint errors: https://github.com/OCA/pylint-odoo/issues/102

**Comment by eLBati** [comment]
Green now, please review

File: l10n_it_fatturapa/__init__.py [hunk 1]
```diff
@@ -1,23 +1,7 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright (C) 2014 Davide Corio <davide.corio@abstract.it>
```
Comment (tafaRU) [inline l10n_it_fatturapa/__init__.py]: Please remove (C) according to the guidelines ;)

File: l10n_it_fatturapa/__manifest__.py [hunk 2]
```diff
@@ -1,34 +1,19 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright (C) 2014 Davide Corio <davide.corio@abstract.it>
```
Comment (tafaRU) [inline l10n_it_fatturapa/__manifest__.py]: Same as above.

File: l10n_it_fatturapa/__manifest__.py [hunk 3]
```diff
@@ -46,8 +31,10 @@
     ],
     "test": [],
```
Comment (tafaRU) [inline l10n_it_fatturapa/__manifest__.py]: Please remove it.

File: l10n_it_fatturapa_out/__init__.py [hunk 4]
```diff
@@ -1,23 +1,7 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright (C) 2014 Davide Corio
```
Comment (tafaRU) [inline l10n_it_fatturapa_out/__init__.py]: Same as above.

File: l10n_it_fatturapa_out/__manifest__.py [hunk 5]
```diff
@@ -1,29 +1,15 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright (C) 2014 Davide Corio
```
Comment (tafaRU) [inline l10n_it_fatturapa_out/__manifest__.py]: Same as above.

File: l10n_it_fatturapa_out/__manifest__.py [hunk 6]
```diff
@@ -37,7 +23,7 @@
         'security/ir.model.access.csv',
     ],
     "test": [],
```
Comment (tafaRU) [inline l10n_it_fatturapa_out/__manifest__.py]: Please remove it

File: l10n_it_fatturapa_out/models/__init__.py [hunk 7]
```diff
@@ -1,23 +1,7 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright (C) 2014 Davide Corio
```
Comment (tafaRU) [inline l10n_it_fatturapa_out/models/__init__.py]: Same as above.

File: l10n_it_fatturapa_out/models/account.py [hunk 8]
```diff
@@ -1,31 +1,15 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright (C) 2014 Davide Corio
```
Comment (tafaRU) [inline l10n_it_fatturapa_out/models/account.py]: Same as above.

File: l10n_it_fatturapa_out/models/attachment.py [hunk 9]
```diff
@@ -1,36 +1,19 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright (C) 2014 Davide Corio
```
Comment (tafaRU) [inline l10n_it_fatturapa_out/models/attachment.py]: Same as above.

File: l10n_it_fatturapa_out/tests/__init__.py [hunk 10]
```diff
@@ -1,21 +1,5 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright (C) 2014 Davide Corio
```
Comment (tafaRU) [inline l10n_it_fatturapa_out/tests/__init__.py]: Same as above.

File: l10n_it_fatturapa_out/wizard/__init__.py [hunk 11]
```diff
@@ -1,21 +1,14 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright (C) 2014 Davide Corio
```
Comment (tafaRU) [inline l10n_it_fatturapa_out/wizard/__init__.py]: Same as above.

**Comment by eLBati** [comment]
@tafaRU thanks, done

**Comment by CiroBoxHub** [comment]
:+1:

**Comment by eLBati** [comment]
Aspettiamo la tabella "natura" (modulo base generico)

**Comment by eLBati** [comment]
Eccolo: https://github.com/OCA/l10n-italy/pull/387

File: l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py [hunk 12]
```diff
@@ -584,36 +516,28 @@ def setDettaglioLinee(self, cr, uid, invoice, body, context=None):
...
-            tax_id = self.pool['account.tax'].get_tax_by_invoice_tax(
-                cr, uid, tax_line.name, context=context)
-            tax = tax_pool.browse(cr, uid, tax_id, context=context)
+    def setDatiRiepilogo(self, invoice, body):
+        for tax_line in invoice.tax_line_ids:
+            tax = tax_line.tax_id
             riepilogo = DatiRiepilogoType(
-                AliquotaIVA='%.2f' % (tax.amount * 100),
+                AliquotaIVA='%.2f' % tax.amount,
                 ImponibileImporto='%.2f' % tax_line.base,
```
Comment (eLBati) [inline l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py]: Pare che il campo ```base``` non esista più

Comment (eLBati) [reply l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py]: Non è vero, esiste

**Pull request merged**